platform/core/src/rdf.rs

```diff
@@ -0,0 +1,91 @@
+use std::collections::HashMap;
+use oxigraph::store::{Store, StorageError};
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::rdf_failure::RdfFailure;
+
+pub type SPARQLQuery = String;
+
+/// A simplified serializable replica of oxigraph::sparql::QuerySolution
+#[derive(Deserialize, Serialize, PartialEq, Debug, Clone)]
+pub struct QuerySolution {
+    values: HashMap<String, String>,
+}
+
+impl QuerySolution {
+    fn new(values: HashMap<String, String>) -> Self {
+        Self { values }
+    }
+}
+
+/// A simplified serializable replica of oxigraph::sparql::QueryResults
+/// Only Solutions for SELECT queries are currently supported
+#[derive(Deserialize, Serialize, PartialEq, Debug, Clone)]
+pub enum QueryResults {
+    /// Results of a [SELECT](https://www.w3.org/TR/sparql11-query/#select) query.
+    Solutions(Vec<QuerySolution>),
+    // TODO: Implement those when needed
+    // Result of a [ASK](https://www.w3.org/TR/sparql11-query/#ask) query.
+    // Boolean(bool),
+    // Results of a [CONSTRUCT](https://www.w3.org/TR/sparql11-query/#construct) or [DESCRIBE](https://www.w3.org/TR/sparql11-query/#describe) query.
+    // Graph(QueryTripleIter),
+}
+
+fn init_store(app_path: String) -> Result<Store, StorageError> {
+    Store::open(app_path + env!("RDF_DB_PATH"))
+}
+
+pub fn rdf_query(query: SPARQLQuery, app_path: String) -> Result<QueryResults, RdfFailure> {
+    let store = init_store(app_path).map_err(|err| RdfFailure::failed_to_initialize_store(err))?;
```

Comment:
Nit:
I think you can remove the closure and call map_err as follows:
```
init_store(app_path).map_err(RdfFailure::failed_to_initialize_store)
```
Same for other instances.